crates/core/tedge_agent/src/restart_manager/actor.rs

```diff
@@ -206,17 +207,38 @@ impl RestartManagerActor {
 
     async fn get_restart_operation_commands(&self) -> Result<Vec<Command>, RestartManagerError> {
         let mut vec = vec![];
-        // sync first
-        let mut sync_command = Command::new(SUDO);
-        sync_command.arg(SYNC);
-        vec.push(sync_command);
 
         // reading `config_dir` to get the restart command or defaulting to `["init", "6"]'
         let system_config = SystemConfig::try_new(&self.config.config_dir)?;
 
-        let mut command = Command::new(SUDO);
-        command.args(system_config.system.reboot);
-        vec.push(command);
+        // Check if sudo command is available
+        match Command::new(SUDO)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
```

Comment:
Maybe using the [which crate](https://docs.rs/which/latest/which/) would be a more simple approach which does not involve executing `sudo`. We are already using `which` in the `tedge connect` command.

Reply:
@didier-wenzek What do you think about using `which` instead of calling sudo to test its existence?

Reply:
I agree with @reubenmiller that using the [which crate](https://docs.rs/which/latest/which/) would less brittle.

Reply:
I would even go a step further by adding a `get_sudo_path() -> Option<PathBuf>` function in `tedge_utils`.

Reply:
Replaced with `which` crate